### PR TITLE
Add Swagger documentation

### PR DESCRIPTION
**Pull request by orkosinha**
## Overview

Adds documentation for all of the current routes.

## Changes Made

Added the Swagger UI package, `DocRouter` for rendering docs, and `swagger.json` for all of the documentation content. There was another issue with `ApplicationRouter` where not all middleware is a type of `ExpressCallback` as of a recent Express version. Also updated the README. 

## Test Coverage

Tested by running the documentation.

### CODE REVIEW

File: src/routers/DocsRouter.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Request } from 'express';
+import ApplicationRouter from '../utils/ApplicationRouter';
+import * as swaggerUI from 'swagger-ui-express';
+import * as swaggerDocument from '../swagger.json';
+import { ExpressCallback } from '../utils/ApplicationAPI';
```
Comment (seancorc) [inline src/routers/DocsRouter.ts]: Delete unused import of { ExpressCallback }

File: src/swagger.json [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+        "responses": {
+          "200": {
+            "description": "{success: true, data: Matching}"
+          },
+          "401": {
+            "description": "{success: false, { errors: [string] }}"
+          }
+        }
+      }
+    },
```
Comment (seancorc) [inline src/swagger.json]: This route is used for creating matchings, not getting them.

**Review by seancorc — CHANGES_REQUESTED** [review]
Looks awesome Orko!! 💯 💯 

Just some minor comments and thoughts:
- Is `/auth/` the best prefix for the general routes?
- Since I am planning to make a PR soon to change the onboarding flow with respect to the backend I will probably change a few things in that PR.

File: README.md [hunk 3]
```diff
@@ -1 +1,7 @@
-# coffee-chats-backend
\ No newline at end of file
+# Pear Backend
+
+To start Pear Backend, run `npm install` and `npm run start`.
+
+Pear Backend has a linter which can be used with `npm run check` to view issues, and `npm run fix` to auto-fix some of those issues.
+
+Documentation on routes and usage is [here](https://pear-backend.cornellappdev.com/api/v1/docs/).
```
Comment (leewang-codes) [inline README.md]: maybe add a line on setting up the database and environment variables!

**Review by leewang-codes — APPROVED** [review]
nice!

**Review by seancorc — APPROVED** [review]

**Pull request merged**